btx/processing/pipca.py

```diff
@@ -175,6 +175,76 @@ def run_model(self):
                 f.create_dataset('V', data=V)
                 print(f'Model saved to {self.filename}')
 
+    def run_model_full(self, previous_U, previous_S, previous_mu, previous_var):
```

Comment:
Likewise, there is a very high level of redundancy with `run_model`. Consider merging them into one.

Reply:
I merged the two functions:

```
def run_model_full(self, previous_U = None, previous_S = None, previous_mu = None, previous_var = None):
        """
        Perform iPCA on run subject to initialization parameters.
        """
        batch_size = self.batch_size
        num_images = self.num_images

        # initialize and prime model, if specified
        if self.priming:
            img_batch = self.get_formatted_images(
                self.num_components, 0, self.num_features
            )
            self.prime_model(img_batch)

        elif previous_U is not None:
            self.U = previous_U
            self.S = previous_S
            self.mu = previous_mu
            self.total_variance = previous_var

        else:
            self.U = np.zeros((self.split_counts[self.rank], self.num_components))
            self.S = np.ones(self.num_components)
            self.mu = np.zeros((self.split_counts[self.rank], 1))
            self.total_variance = np.zeros((self.split_counts[self.rank], 1))

        # divide remaining number of images into batches
        # will become redundant in a streaming setting, need to change
        rem_imgs = num_images - self.num_incorporated_images
        batch_sizes = np.array(
            [batch_size] * np.floor(rem_imgs / batch_size).astype(int)
            + ([rem_imgs % batch_size] if rem_imgs % batch_size else [])
        )

        # define batch indices based on batch sizes
        self.batch_indices = self.distribute_images_over_batches(batch_sizes)
        self.batch_number = 0

        # update model with remaining batches
        for batch_size in batch_sizes:
            self.fetch_and_update_model(batch_size)
            
        self.comm.Barrier()
        
        U = self.gather_U()
        S = self.S
        V = self.V
        
        if self.rank == 0:  
            print("Model complete")
        
            # save model to an hdf5 file
            with h5py.File(self.filename, 'w') as f:
                f.create_dataset('exp', data=self.psi.exp)
                f.create_dataset('run', data=self.psi.run)
                f.create_dataset('det_type', data=self.psi.det_type)
                f.create_dataset('start_offset', data=self.start_offset)
                f.create_dataset('loadings', data=self.pc_data)
                f.create_dataset('U', data=U)
                f.create_dataset('S', data=S)
                f.create_dataset('V', data=V)
                f.create_dataset('mu', data=self.mu)
                f.create_dataset('total_variance', data=self.total_variance)
                print(f'Model saved to {self.filename}')
```